### DISCUSSION TITLE
control.extended.goToPosition

### DISCUSSION THREAD

**Discussion by m4lt3-87** [mrbungle64/ioBroker.ecovacs-deebot#140] [Q&A] [answered]
Hallo,
wie kann die gotoposition als Button in eine VIS eingetragen werden. 
Ich habe den Robbi an einer Position während der Reinigung gestoppt und mir den Positions-Wert notiert. Nun würde ich gerene für die Filterleerung den Robi nach abgeschlossenem Reinigungsvorgang per VIS Button-Druck an die Stelle fahren lassen. Wie kann ich das Umsetzten? (sorry, bin noch relativ neu auf dem Gebiet).

**Comment by m4lt3-87** [comment, marked as answer]
ok, kann es mir nun selsbt beantworten. Der angegebene Wert darf nur 2-stellig sein. Der dritte Wert (Winkel) muss entfallen. Dann funktioniert es.

**Reply by mrbungle64** [reply]
Hallo @m4lt3-87 

ich habe [hier im Wiki](https://github.com/mrbungle64/ioBroker.ecovacs-deebot/wiki/Datenpunkte-%28DE%29) die Datenpunkte jeweils kurz beschrieben.
Falls Du noch Fragen hast kannst Du das gerne hier noch tun.

Zur VIS kann ich aber nicht viel sagen, da ich die selbst gar nicht nutze.
Wenn Du zur VIS Fragen hast wäre das [ioBroker Forum](https://forum.iobroker.net/) wahrscheinlich geeigneter.

**Comment by ossisp1** [comment]
Hallo nochmal zu dem Thema. Ich hab einen Deebot T9 Aivi. 
Wann kann die GotoPosition aktivert werden?
Zu jeder Zeit oder muß der Sauger auf Pause oder Stop sein?

**Reply by mrbungle64** [reply]
@ossisp1 

Bei meinem T8 AIVI geht es jederzeit - dann kann aber beim T9 AIVI wieder anders sein.

**Reply by mrbungle64** [reply]
Es kann sein, dass es bei der letzten Stable (1.4.8) noch so war, dass man die Option "Native "goToPosition" Funktion benutzen (z.B. T8/T9/N8/X1 Serie)" noch explizit aktivieren musste. Bei aktuellen Version funktioniert es auch mit der Standard-Einstellung ("Vorauswahl verwenden")

**Reply by mrbungle64** [reply]
@ossisp1 

Ist Deine Frage beantwortet?

**Comment by ossisp1** [comment]
Ich hab's gestern nochmals versucht. Er will starten bricht dann aber mit
Fehler ab und fährt direkt wieder zur Ladestation. Ich muß das heute
nochmal ausführlicher testen.

Sascha ***@***.***> schrieb am Sa., 28. Jan. 2023, 09:09:

**Reply by mrbungle64** [reply]
> Er will starten bricht dann aber mit Fehler ab und fährt direkt wieder zur Ladestation

Das hört sich danach an, dass die Position ungültig ist oder blockiert ist.

> bricht dann aber mit Fehler ab

Wie äußert sich der Fehler und was steht dazu im Log?

**Reply by ossisp1** [reply]
Bin endlich dazu gekommen wieder zu testen.
Jetzt funktioniert es einwandfrei. 
Als nächstes möchte ich gerne das er am Ende der Reinigung die Position am Mülleimer anfährt. Welcher Datenpunkt sagt mir "Reinigung beendet"? Wenn ich den Status nehme steht da nur "returning" und darauf kann ich ja nicht triggern.

**Reply by mrbungle64** [reply]
@ossisp1 

> Als nächstes möchte ich gerne das er am Ende der Reinigung die Position am Mülleimer anfährt. Welcher Datenpunkt sagt mir "Reinigung beendet"? Wenn ich den Status nehme steht da nur "returning" und darauf kann ich ja nicht triggern.

Es gibt dafür keinen Datenpunkt.
Du kannst aber beim Datenpunkt "status.device" auf den Statuswechsel von "cleaning" auf "returning" triggern.

**Reply by ossisp1** [reply]
Ok. Dankeschön. Und danke für den Adapter.
